redbot/cogs/mod/kickban.py

```diff
@@ -70,48 +70,81 @@ async def ban_user(
+        removed_temp = False
+
+        if not (0 <= days <= 7):
             return _("Invalid days. Must be between 0 and 7.")
 
-        toggle = await self.config.guild(guild).dm_on_kickban()
-        if toggle:
-            with contextlib.suppress(discord.HTTPException):
-                em = discord.Embed(
-                    title=bold(_("You have been banned from {guild}.").format(guild=guild))
+        if isinstance(user, discord.Member):
+            if author == user:
+                return _("I cannot let you do that. Self-harm is bad {}").format(
+                    "\N{PENSIVE FACE}"
                 )
-                em.add_field(
-                    name=_("**Reason**"),
-                    value=reason if reason is not None else _("No reason was given."),
-                    inline=False,
+            elif not await is_allowed_by_hierarchy(self.bot, self.config, guild, author, user):
+                return _(
+                    "I cannot let you do that. You are "
+                    "not higher than the user in the role "
+                    "hierarchy."
                 )
-                await user.send(embed=em)
+            elif guild.me.top_role <= user.top_role or user == guild.owner:
+                return _("I cannot do that due to Discord hierarchy rules.")
+
+            toggle = await self.config.guild(guild).dm_on_kickban()
+            if toggle:
+                with contextlib.suppress(discord.HTTPException):
+                    em = discord.Embed(
+                        title=bold(_("You have been banned from {guild}.").format(guild=guild))
+                    )
+                    em.add_field(
+                        name=_("**Reason**"),
+                        value=reason if reason is not None else _("No reason was given."),
+                        inline=False,
+                    )
+                    await user.send(embed=em)
+
+            ban_type = "ban"
+        else:
+            tempbans = await self.config.guild(guild).current_tempbans()
+
+            ban_list = await guild.bans()
+            for entry in ban_list:
+                if entry.user.id == user.id:
+                    if user.id in tempbans:
+                        async with self.config.guild(guild).current_tempbans() as tempbans:
+                            tempbans.remove(user.id)
+                        removed_temp = True
+                    else:
+                        return _("User {user_id} is already banned.").format(user_id=user.id)
+
+            ban_type = "hackban"
 
         audit_reason = get_audit_reason(author, reason)
 
         queue_entry = (guild.id, user.id)
         try:
             await guild.ban(user, reason=audit_reason, delete_message_days=days)
-            log.info(
-                "{}({}) banned {}({}), deleting {} days worth of messages.".format(
-                    author.name, author.id, user.name, user.id, str(days)
+            if removed_temp == True:
```

Comment:
We mostly adhere to PEP 8 style guide, so this should just be using `if greeting:` without `==`.
````suggestion
            if removed_temp:
```